### ISSUE TITLE
Arrays in original target are mutated by library

### ISSUE THREAD

**Issue by MZanggl** [ElliotNB/observable-slim#43]
When you have this code 

```typescript
const item = { numbers: [1, 2] }
ObservableSlim.create(item)
```

calling `Object.keys(item.numbers)` yields `['0', '1', '__length']` instead of just `['0', '1']`.

This is due to the following change: https://github.com/ElliotNB/observable-slim/commit/c69aafd14ca851b63ac76262d951b9d057fa2929#diff-a882dd68fa93398bef8261d2308d9600efd956dbe6ba6ef1f0fe368aa107225dR64

Expected: Original object should not be mutated.

Of course `Object.keys()` is not really useful on arrays, but it causes a strange error in jest saying `serializes to the same string`.

**Comment by ElliotNB** [comment]
@MZanggl I don't think I can remove `__length` entirely without compromising other functionality, but I can set `__length` so that it is both immutable and not enumerable. For example:

```
var test = [1,2];
Object.defineProperty(test, "__length", {enumerable:false, value:2});
console.log(Object.keys(test));    // logs ["0","1"] to the console
console.log(test.__length);    // logs 2 to the console
```

Would that resolve your particular use case?

**Comment by MZanggl** [comment]
@ElliotNB Oh yes, that would work! Tested it in a sandbox: https://codesandbox.io/s/jest-test-forked-g1oqv?file=/index.test.js

**Comment by ElliotNB** [comment]
@MZanggl Excellent! I'll get a PR opened up for it and get it published out on npm soon.

**Comment by BinarySpike** [comment]
I need to test this on my side—it may be an improvement for me.  There are differences between the native proxy and the polyfill proxy when it comes to `__length` and in my client code I have to manually filter it out to support IE11 / polyfill.  This might remove the need for my manual check.

I can see no downsides for _my_ use-cases!

**Comment by ElliotNB** [comment]
Resolved via pull request #49 courtesy of @cdbishop
